dataset/slice.cpp

```diff
@@ -32,15 +32,15 @@ scipp::index get_index(const VariableConstView &coord, const Dim dim,
   return std::clamp<scipp::index>(0, i, coord.dims()[dim]);
 }
 
-auto get_1d_coord(const DataArrayConstView &data, const Dim dim) {
+template <typename T> auto get_1d_coord(const T &data, const Dim dim) {
   const auto &coord = data.coords()[dim];
   if (coord.dims().ndim() != 1)
     throw except::DimensionError(
         "Multi-dimensional coordinates cannot be used for value-base slicing.");
   return coord;
 }
 
-auto get_coord(const DataArrayConstView &data, const Dim dim) {
+template <typename T> auto get_coord(const T &data, const Dim dim) {
```

Comment:
I suppose the templating becomes unnecessary if we simply pass `data.coord()` (`CoordsConstView`) instead of `data` to these two functions?